whois-api/src/main/java/net/ripe/db/whois/api/rdap/WhoisRdapService.java

```diff
@@ -215,14 +215,18 @@ private Response lookupWithRedirectUrl(final HttpServletRequest request, final S
     }
 
     private Response lookupForAutNum(final HttpServletRequest request, final String key) {
-        if (isRedirect(AUT_NUM, key) && !rdapRequestValidator.isReservedAsNumber(key)) {
-            return redirect(getRequestPath(request), AUT_NUM, key);
-        }
+        try {
+            if (isRedirect(AUT_NUM, key) && !rdapRequestValidator.isReservedAsNumber(key)) {
+                return redirect(getRequestPath(request), AUT_NUM, key);
+            }
 
-        final Query query = getQueryObject(ImmutableSet.of(AUT_NUM), key);
-        List<RpslObject> result = rdapQueryHandler.handleAutNumQuery(query, request);
+            final Query query = getQueryObject(ImmutableSet.of(AUT_NUM), key);
+            List<RpslObject> result = rdapQueryHandler.handleAutNumQuery(query, request);
 
-        return getResponse(request, result);
+            return getResponse(request, result);
+        } catch (RdapException ex){
+            throw new AutnumException(ex.getErrorTitle(), ex.getErrorDescription(), ex.getErrorCode());
```

Comment:
Are you sure onl Rdapexception are being thrown , I see mapping for lot other exception like paramException which are not being turned into RdapException so on those situations we are still not sending autnum exception? I think we should turn all exceptions into rdapExceptions to make it work properly

Reply:
paramException is thrown by jetty I think.
We don't have control on all the exception that we can thrown